CSharpSourceCode/AttributeDataSystem/StaticAttributeMissionLogic.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System.Collections.Generic;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.Core;
+using TaleWorlds.MountAndBlade;
+
+namespace TOW_Core.AttributeDataSystem
+{
+    /// <summary>
+    /// Reads out PartyAttributeManager during Combat
+    /// </summary>
+    public class StaticAttributeMissionLogic : MissionLogic
+    {
+        private Mission.TeamCollection teams;
+        private PartyAttributeManager _partyAttributeManager;
+
+        private List<PartyAttribute> activeAttributes;
+        
+        private List<PartyAttribute> defenderAttributes;
+
+        private List<Agent> _agents;
+        
+        public override void AfterStart()
+        {
+            base.AfterStart();
+            teams= Mission.Current.Teams;
+            activeAttributes = new List<PartyAttribute>();
+            defenderAttributes = new List<PartyAttribute>();
+            _agents = new List<Agent>();
+            
+            if (Campaign.Current != null)
+            {
+                _partyAttributeManager= PartyAttributeManager.Instance;
+                activeAttributes = _partyAttributeManager.GetActiveInvolvedParties();
+            }
+            
+        }
+        
+         public override void OnAgentBuild(Agent agent, Banner banner)
+        {
+            base.OnAgentBuild(agent, banner);
+
+            if (agent.IsMount)
+                return;
+            
+            foreach (var partyAttribute in activeAttributes)
+            {
+                if (agent.Origin.BattleCombatant== partyAttribute.PartyBase)
+                {
+                    if (partyAttribute.PartyType == PartyType.RogueParty)
+                    {
+                        foreach (var attribute in partyAttribute.RegularTroopAttributes)
+                        {
+                            AddStaticAttributeComponent(agent, attribute, partyAttribute);
+                            break;
```

Comment:
Why do all of these foreach loops break after the first iteration? If it really does need to break, can't you just get rid of the loop entirely?